File: src/leetcode/80_remove-duplicates-from-sorted-array-ii/20251030.go
func removeDuplicates(nums []int) int {
    l := len(nums) 
    if l <= 2 {
        return l
    }

    k := 2

    for i := 2; i < l; i++ {
        if nums[k - 2] != nums[i] {
            nums[k] = nums[i]
            k++
        }
    }
    
    return k;
}

File: src/leetcode/169_majority-element/20251030.ts
export function majorityElement(nums: number[]): number {
    if (nums.length <= 2) {
        return nums[0]
    }

    const arr1 = [nums[0]]
    const arr2 = [] 
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== arr1[0]) {
            arr2.push(nums[i])
        } else {
            arr1.push(nums[i])
        }
    }

    if (arr1.length > nums.length / 2) {
        return arr1[0]
    } else {
        return arr2[0]
    }
};
